Azure/DASH
# PTF: add basic test infra and more test cases

This patch-set introduce PTF test cases and basic infrastructure (similar to SONiC PTF) to run PTF DASH tests. For now the test cases supports minimal set of use cases, but it is the base for future test case implementations that will cover whole overlay PTF test scenario (in progress).

**Main changes:**

- Add basic PTF infra (copied from SONiC PTF) for running PTF, allows run test cases w/o SAI repo, only PTF is required. DASH repo becomes main repo for test development.
- Add Common VNet API for all tests (to void duplicate work, can be split from this file later)
- Add the following test cases:
  - Inbound VNI/ENI
  - Inbound Routing PA validate
  - Inbound Routing VNI match
  - Inbound Routing PA SRC IP
  - Outbound Routing Vnet direct
  - Outbound Routing direct
  - CT: just placeholder, can be used for debugging only
  - L3 Route basic: just to make sure traffic is routed
- Run only bmv2 ready test case on CI because of missing functionality in bmv2. Once more bmv2 functionality becomes available, test_saithrift_vnet.py can be merged with this one
- test_saithrift_vnet: removed unused libs

**Future plan:**
- Topology/setup diagram
- Test plan (in progress)
- Check/validate BMV2 support
- More CI integration
- More updates and fixes
- Update TCs to SAI PTF repo (once stable)
